StyleCop.Analyzers/StyleCop.Analyzers/MaintainabilityRules/SA1400CodeFixProvider.cs

```diff
@@ -115,14 +115,17 @@ public override FixAllProvider GetFixAllProvider()
 
                 if (updatedDeclarationNode != null)
                 {
-                    var syntaxRoot = await context.Document.GetSyntaxRootAsync(context.CancellationToken);
-                    var newSyntaxRoot = syntaxRoot.ReplaceNode(declarationNode, updatedDeclarationNode);
-                    var newDocument = context.Document.WithSyntaxRoot(newSyntaxRoot);
-                    context.RegisterCodeFix(CodeAction.Create("Declare accessibility", token => Task.FromResult(newDocument)), diagnostic);
+                    context.RegisterCodeFix(CodeAction.Create("Declare accessibility", token => GetTransformedDocument(context, root, declarationNode, updatedDeclarationNode)), diagnostic);
```

Comment:
:exclamation: Move as much of the logic from this loop to the callback as possible. Most of the work is still being done in the main loop.